libpod/container_internal_common.go

```diff
@@ -365,13 +369,29 @@ func (c *Container) generateSpec(ctx context.Context) (s *spec.Spec, cleanupFunc
 				if err := c.relabel(m.Source, c.MountLabel(), label.IsShared(o)); err != nil {
 					return nil, nil, err
 				}
-
+			case "preserve_symlinks":
+				fi, err := os.Lstat(m.Source)
+				if err != nil {
+					return nil, nil, err
+				}
+				if fi.Mode()&fs.ModeSymlink == os.ModeSymlink {
+					target, err := os.Readlink(m.Source)
+					if err != nil {
+						return nil, nil, fmt.Errorf("presving symlink: reading symlink: %w", err)
+					}
+					createSymlinks[m.Destination] = target
+					addMount = false
+				}
```

Comment:
Should this return an error if the file is not a symlink?

Reply:
That would break glob mounts where each match is a dedicated single bind mount.  Something that needs explanation in the docs.